src/dodal/plans/wrapped.py

```diff
@@ -0,0 +1,42 @@
+from typing import Annotated, Any
+
+import bluesky.plans as bp
+from bluesky.protocols import Readable
+from pydantic import Field, NonNegativeFloat, validate_call
+
+from dodal.common import MsgGenerator
+from dodal.plan_stubs.data_session import attach_data_session_metadata_decorator
+
+
+@attach_data_session_metadata_decorator()
+@validate_call(config={"arbitrary_types_allowed": True})
+def count(
+    detectors: Annotated[
+        set[Readable],
+        Field(
+            description="Set of readable devices, will take a reading at each point",
+            min_length=1,
+        ),
+    ],
+    num: Annotated[int, Field(description="Number of frames to collect", ge=1)] = 1,
+    delay: Annotated[
+        NonNegativeFloat | list[NonNegativeFloat],
+        Field(
+            description="Delay between readings: if list, len(delay) == num - 1 and \
+            the delays are between each point, if value or None is the delay for every \
+            gap",
+            json_schema_extra={"units": "s"},
+        ),
+    ] = 0.0,
+    metadata: dict[str, Any] | None = None,
+) -> MsgGenerator:
+    """Reads from a number of devices.
+    Wraps bluesky.plans.count(det, num, delay, md=metadata) exposing only serializable
+    parameters and metadata."""
```

Comment:
Nit: I assume we've discussed with bluesky to try and get this into the core codebase and couldn't? May be worth a link to that discussion in case anyone asks the question again

Reply:
It was a series of mostly offline discussions and should probably be written up as an ADR

Reply:
Does someone involved in the discussion have time to write an issue with some back of the envelope bullet point reasons and saying to put them in an ADR, just before it's lost to time?

Reply:
I believe we need https://github.com/DiamondLightSource/blueapi/issues/474 for the @attach_metadata to be handled in blueapi consistently, then that decorator gets moved back into blueapi and then blueapi can just use bluesky.plans as a plans module (or... the ones we actually want).

Reply:
In which case can we have a comment here saying we can remove this when that has been done?

Reply:
I think actually we'll need to maintain them for at least a while afterwards- bp.count has a an argument of type `PerShot` which is not serialisable. We may be able to ignore all fields that are optional and of a non-serialisable type? Especially where they are `Callable`s. I'll add something to that effect.

```python
PerShot = Callable[[Sequence[Readable], Optional[bps.TakeReading]], MsgGenerator]
``` 

https://github.com/bluesky/bluesky/blob/92d2633127e33e51aa1d1b109c80503e74bbe0e0/src/bluesky/plans.py#L66